### PR TITLE
Added SystemObject type

### PR DESCRIPTION
**Pull request by ncave**
@alfonsogarciacaro You don't really have to merge this if you're not ok with it, but perhaps it can help reason about complicated corner cases for comparing and equality of deeply nested types. This is passing tests, but may need more work if corner cases require it.

### CODE REVIEW

File: src/dotnet/Fable.Compiler/Transforms/Fable2Babel.fs [hunk 1]
```diff
@@ -1369,6 +1371,8 @@ module Util =
             | Some(TransformExpr com ctx baseRef) -> yield inherits com ctx consIdent baseRef
             | None when info.Entity.IsValueType ->
                 yield coreValue com ctx "Types" "Record" |> inherits com ctx consIdent
+            | None when info.Entity.IsClass ->
+                yield coreValue com ctx "Types" "SystemObject" |> inherits com ctx consIdent
```
Comment (alfonsogarciacaro) [inline src/dotnet/Fable.Compiler/Transforms/Fable2Babel.fs]: I guess we need to check if the class is not already inheriting anything else.

File: src/dotnet/Fable.Compiler/Transforms/Replacements.fs [hunk 2]
```diff
@@ -601,10 +597,6 @@ let rec equals (com: ICompiler) r equal (left: Expr) (right: Expr) =
         Helper.CoreCall("Reflection", "equals", Boolean, [left; right], ?loc=r) |> is equal
     | Tuple _ ->
         Helper.CoreCall("Util", "equalArrays", Boolean, [left; right], ?loc=r) |> is equal
-    | DeclaredType(ent,_) when hasBaseImplementingBasicMethods ent ->
-        Helper.InstanceCall(left, "Equals", Boolean, [right]) |> is equal
```
Comment (alfonsogarciacaro) [inline src/dotnet/Fable.Compiler/Transforms/Replacements.fs]: Maybe we can leave this for optimization? It'll be faster to call `.Equals` directly.

File: src/dotnet/Fable.Compiler/Transforms/Fable2Babel.fs (hunk 1 above)
Comment (alfonsogarciacaro) [reply src/dotnet/Fable.Compiler/Transforms/Fable2Babel.fs]: Sorry, scratch that. I thought this belonged to another file 😅

**Comment by alfonsogarciacaro** [comment]
No problem, let's merge this and just assume types coming from JS may have different equality semantics :) As now all types will be deriving from something, I've added a `fable-core/Types.declare` helper so we can have better control of the inheritance.

I'm thinking that we could announce a Fable 2 beta so people can start playing with it, what do you think @ncave? Better to wait until the REPL works and we don't need to change many design decisions?

File: src/dotnet/Fable.Compiler/Transforms/Replacements.fs (hunk 2 above)
Comment (ncave) [reply src/dotnet/Fable.Compiler/Transforms/Replacements.fs]: Right, same for classes.

**Comment by ncave** [comment]
@alfonsogarciacaro Go for it, unless you want to do one last alpha.
